public/services/CampaignsApi.js

```diff
@@ -38,11 +39,51 @@ export function fetchLatestAnalytics() {
   });
 }
 
+async function fetchShareCounts(analytics) {
+  const paths = Object.entries(analytics.analyticsByPath || {});
+
+  // Facebook shares should be fetched sequentially to avoid rate-limit.
+  for (const [key, values] of paths) {
+    const req = {
+      url: `https://graph.facebook.com/?id=https://theguardian.com${key}`,
+      method: 'get'
+    };
+    const fbRateLimit = new Promise(function(resolve) {
+      Reqwest(req)
+        .then(res => {
+          values.facebookShares = res.share ? res.share.share_count : 0;
+          resolve();
+        })
+        .fail(resolve);
+    });
+    await fbRateLimit;
+  }
+  // LinkedIn shares should be fetched in parallel.
+  const linkedInShares = paths.map( ([key, values]) => {
+    const req = {
+      url: `https://www.linkedin.com/countserv/count/share?url=https://theguardian.com${key}`,
+      type: 'jsonp'
+    };
+    return new Promise(function(resolve) {
+      Reqwest(req)
+        .then(res => {
+          values.linkedInShares = res.count;
+          resolve();
+        })
+        .fail(resolve);
+    });
+  });
+  return Promise.all(linkedInShares)
```

Comment:
Wouldn't it be better to fetch the share counts in a batch job and cache them in dynamo?

Reply:
I'm not that worried about performance, and the information is coming from an API we have direct access to. It seems simple to just access it?

Reply:
I thought there was a requirement to report on shares by week etc but if not it's not a problem.  Could be done later anyway if needed.

Reply:
@rich-nguyen @kelvin-chappell Maybe worth checking with Chloe or Hannah.